### PR TITLE
Shibboleth login - Added email validation

### PR DESCRIPTION
**Pull request by milanmajchrak**
| Phases            | MP | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |  0  |    0 |     0 |      0 |        0 |
| Developing      |  0  |  0  |    0 |    0 |      0 |         0 |
| Review             |  0  |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -  |   -   |  -    |   -    |         0 |
| ETA est.             |      |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -  |   -   |   -     |        0 |

### CODE REVIEW

**Review by MajoBerger — APPROVED** [review]

File: dspace-api/src/main/java/org/dspace/authenticate/clarin/ClarinShibAuthentication.java [hunk 1]
```diff
@@ -1324,5 +1329,12 @@ public boolean canChangePassword(Context context, EPerson ePerson, String curren
     public boolean areSpecialGroupsApplicable(Context context, HttpServletRequest request) {
         return true;
     }
+
+    public String getEmailAcceptedOrNull(String email) {
+        if (email == null || email.isEmpty() || email.matches(".*\\s+.*")) { // no whitespaces in mail
```
Comment (vidiecan) [inline dspace-api/src/main/java/org/dspace/authenticate/clarin/ClarinShibAuthentication.java:1334]: /\s/.test(email); ?

**Review by vidiecan — CHANGES_REQUESTED** [review]

File: dspace-api/src/main/java/org/dspace/authenticate/clarin/ClarinShibAuthentication.java (hunk 1 above)
Comment (vidiecan) [reply dspace-api/src/main/java/org/dspace/authenticate/clarin/ClarinShibAuthentication.java:1334]: isn't `/\s/.test(email)`  better?

**Review by vidiecan — CHANGES_REQUESTED** [review]

File: dspace-api/src/main/java/org/dspace/authenticate/clarin/ClarinShibAuthentication.java (hunk 1 above)
Comment (milanmajchrak) [reply dspace-api/src/main/java/org/dspace/authenticate/clarin/ClarinShibAuthentication.java:1334]: Maybe yes, but I don't understand that code. What is .test(email)? Some method?

Comment (vidiecan) [reply dspace-api/src/main/java/org/dspace/authenticate/clarin/ClarinShibAuthentication.java:1334]: regex match vs search

**Review by vidiecan — CHANGES_REQUESTED** [review]

File: dspace-api/src/main/java/org/dspace/authenticate/clarin/ClarinShibAuthentication.java (hunk 1 above)
Comment (milanmajchrak) [reply dspace-api/src/main/java/org/dspace/authenticate/clarin/ClarinShibAuthentication.java:1334]: The code `/\\s/.test(email)` is written in JavaScript and it checks if the given email string contains any whitespace character. 

In Java it looks like this: `Pattern.compile("\\s").matcher(email).find()` 
I changed it to that Java Pattern condition.

**Review by vidiecan — APPROVED** [review]

**Pull request merged**